src/ly_wrap.c

```diff
@@ -398,7 +398,7 @@ sr_lys_find_xpath(const struct ly_ctx *ctx, const char *xpath, uint32_t options,
         *valid = 1;
     }
 
-    if (lys_find_xpath(ctx, NULL, xpath, options, set)) {
+    if (lys_find_xpath(ctx, NULL, xpath, options, set) || !(*set) || !(*set)->count) {
```

Comment:
The rule should be that either `lys_find_xpath()` returns an error or `set` will be non-`NULL` (but may still be empty). So the `*set` check (and in your other changes) should be redundant.